core/src/main/java/org/apache/iceberg/BaseOverwriteFiles.java

```diff
@@ -95,11 +101,18 @@ public OverwriteFiles caseSensitive(boolean isCaseSensitive) {
   @Override
   public OverwriteFiles validateNoConflictingAppends(Expression newConflictDetectionFilter) {
     Preconditions.checkArgument(newConflictDetectionFilter != null, "Conflict detection filter cannot be null");
-    this.conflictDetectionFilter = newConflictDetectionFilter;
+    this.appendConflictDetectionFilter = newConflictDetectionFilter;
     failMissingDeletePaths();
     return this;
   }
 
+  @Override
+  public OverwriteFiles validateNoConflictingDeleteFiles(Expression newConflictDetectionFilter) {
+    Preconditions.checkArgument(newConflictDetectionFilter != null, "Conflict detection filter cannot be null");
```

Comment:
Nit: Same comment about saying "Delete conflict detection filter cannot be null` instead of leaving it unqualified and ambiguous.